packages/assistify-deepl-translation/server/deeplTranslate.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @author Vigneshwaran Odayappan <vickyokrm@gmail.com>
+ */
+
+
+import {TranslationProviderRegistry, AutoTranslate} from 'meteor/rocketchat:autotranslate';
+import {RocketChat} from 'meteor/rocketchat:lib';
+import {SystemLogger} from 'meteor/rocketchat:logger';
+import _ from 'underscore';
+
+/**
+ * Represents DEEPL translate class
+ * @class
+ * @augments AutoTranslate
+ */
+class DeeplAutoTranslate extends AutoTranslate {
+	/**
+	 * setup api reference to deepl translate to be used as message translation provider.
+	 * @constructor
+	 */
+	constructor() {
+		super();
+		this.name = 'deepl-translate';
+		this.apiEndPointUrl = 'https://api.deepl.com/v1/translate';
+		// self register & de-register callback - afterSaveMessage based on the activeProvider
+		RocketChat.settings.get('AutoTranslate_ServiceProvider', (key, value) => {
+			if (this.name === value) {
```

Comment:
What for?

Reply:
This will to set the active provide context when the provider settings changed.